src/kdc/do_tgs_req.c

```diff
@@ -290,12 +290,21 @@ process_tgs_req(krb5_kdc_req *request, krb5_data *pkt,
         goto cleanup;
     if (s4u_x509_user != NULL) {
         setflag(c_flags, KRB5_KDB_FLAG_PROTOCOL_TRANSITION);
-        if (is_referral) {
-            /* The requesting server appears to no longer exist, and we found
-             * a referral instead.  Treat this as a server lookup failure. */
-            errcode = KRB5KDC_ERR_S_PRINCIPAL_UNKNOWN;
-            status = "LOOKING_UP_SERVER";
-            goto cleanup;
+
+        if (!is_cross_tgs_principal(header_ticket->server)) {
+            /* If the tgt is local, then both the impersonated
+             * client and the server must be local */
+            if (is_referral || client == NULL) {
```

Comment:
Combine this with the parent `if` conditional

Reply:
That would change the meaning of the 'else' to include the case of local tgt && (!is_referral  && client  != NULL), which is a valid case that I think will break.

Apart of that, I think I better understand the previous comment, and will rather preserve the error for the is_referral case.